### ISSUE TITLE
pragma disable from generated partial class is not applying to when the class is used

### ISSUE THREAD

**Issue by ayuksekkaya** [meziantou/Meziantou.Analyzer#820]
### Version of the Meziantou.Analyzer NuGet package

2.0.202

### Rule Identifier

MA0097

### Target Framework

Net 9.0

### C# Language version

C# 13

### Description

I am using this library called [Vogen](https://github.com/SteveDunn/Vogen) for generating some structs. The way it works is that, you define your struct like this:

```csharp
[ValueObject<Guid>]
public readonly partial struct TenantId;
```
And the library source generates some code. 

 And the auto-generated code using that library has this line at the beginning of the file: 

```csharp
// Suppress Error MA0097 : A class that implements IComparable<T> or IComparable should override comparison operators
#pragma warning disable MA0097
```

However, this isn't working. I still have to manually add  "#pragma warning disable MA0097" to where I am defining the structs in my code.

At first, I thought this was an issue with Vogen and I opened an issue on their GitHub as well here: https://github.com/SteveDunn/Vogen/issues/814

However, I think it might be an issue with the Meziantou analyzers instead, because this is working for other auto-disabled warnings, such as "CS0618", "CA1036", and "CS8669"...

Any thoughts on why this may be happening?

### Reproduction Steps

Minimal code:

```c#
code to reproduce the error
```
Install the Vogen library and just add this line to any file:

[ValueObject<Guid>]
public readonly partial struct TenantId;

**Comment by meziantou** [comment]
MA0047 reports 2 diagnostics, one in each file. CA1036 reports 1 diagnostic with additional locations. Maybe that's the trick. I can try to use the same strategy.

That's being said, I think you can use an attribute to suppress the diagnostic (the attribute can be in the generated file):

````c#
[ValueObject<Guid>, SuppressMessage("","MA0097")]
public readonly partial struct TenantId;
````

**Comment by ayuksekkaya** [comment]
@meziantou Thank you for a speedy reply! 

I didn't know that trick with the `SuppressMessage`. That indeed worked. This would be fine for my use case. 

If you think 1 diagnostic with additional locations improves the overall design, that's fine yes.

The suppress message is fine for my use case. I mainly just didn't wanna keep doing pragma disable everywhere, and I was trying to disable from .editorconfig but couldn't find a way to apply it selectively enough. So thank you for that info!

**Comment by SteveDunn** [comment]
Thank you @meziantou . @ayuksekkaya - can I now close the issue in Vogen?
